### PR TITLE
[release/v0.9] Handle cleanup DNS for attachable container

### PR DESCRIPTION
**Pull request by andrewhsu**
backport:
* #1985 Handle cleanup DNS for attachable container
* #1991 Add test for cleanupServiceDiscovery

with cherry-pick of git commits 1c04e19 52a9ab5:
```
$ git cherry-pick -s -x 1c04e19 52a9ab5
```

conflict with file `service_common.go` resolved by adding functions:
* ~`func (c *controller) getLBIndex(sid, nid string, ingressPorts []*PortConfig) int`~
* `func (c *controller) cleanupServiceDiscovery(cleanupNID string)`

### CODE REVIEW

**Comment by andrewhsu** [comment]
@fcrisciani @abhi @mavenugo

File: service_common.go [hunk 1]
```diff
@@ -141,6 +141,39 @@ func newService(name string, id string, ingressPorts []*PortConfig, serviceAlias
 	}
 }
 
+func (c *controller) getLBIndex(sid, nid string, ingressPorts []*PortConfig) int {
```
Comment (mavenugo) [inline service_common.go]: This function is not used by this commit. I dont think adding this new function to resolve the conflict is appropriate.

Comment (andrewhsu) [reply service_common.go]: i've updated the commit a75ba12 to not include `getLBIndex` in `service_common.go`

**Comment by fcrisciani** [comment]
looks good now, thanks

**Comment by mghazizadeh** [comment]
I believe we have e2e tests for this as well that can now be run automated using testkit. @antonybichon17

**Pull request merged**